### DISCUSSION TITLE
Keep focus after content is updated from the server

### DISCUSSION THREAD

**Discussion by matheusbaumgart** [ueberdosis/tiptap#3544] [General Questions]
Hi, how can I keep the editor in focus when the data I just pushed to the server (firestore) comes back and updates the content?

I'm currently debouncing so I do push changes to Firestore every keystroke, but as soon as I push, data comes back, re-renders the editor and that causes the editor to lose focus.

I could useRef and pass `contentRef.current` as the content, so it doesn't re-render when the content changes, but that's causes me to lose the realtime benefit of firestore, which means that a user could overwrite the latest content in another outdated copy (another device).

Any ideas on how to proceed here? I've been stuck for days.

Thank you!

**Comment by fmeinhold** [comment]
in case anyone comes across this: i am simply setting the focus again:
``` 
onUpdate: ({ editor }) => {
    updateFn() // <- my debounced update function 
    editor.commands.focus()
  },
```
